File: Sistem Bilangan/Desimal/DesimalkeOktal.java
import java.util.Scanner;

public class DesimalkeOktal
{
    public static void main (String [] args)
    {
        Scanner scan = new Scanner (System.in);
        System.out.print ("Desimal       : ");
        int desimal = scan.nextInt ();

        StringBuilder oktal = new StringBuilder ();
        while (desimal > 0)
        {
            int sisa = desimal % 8;
            oktal.insert (0, sisa);
            desimal /= 8;
        }

        System.out.print ("Oktal         : " + oktal);
    }
}

File: Sistem Bilangan/Desimal/Desimal-ke-Oktal.c
#include <stdio.h>

int main ()
{
    int desimal = 0;
    char oktal [32];
    int index = 0;

    printf ("Desimal       : ");
    scanf ("%d", &desimal);

    while (desimal > 0)
    {
        int sisa = desimal % 8;
        oktal[index++] = sisa + 48;
        desimal /= 8;
    }

    printf ("Oktal         : ");
    for (int i = index - 1; i >= 0; i--)
        printf("%c", oktal [i]);

    return 0;
}
